libdrgn/dwarf_index.c

```diff
@@ -2357,11 +2413,28 @@ static struct drgn_error *index_cu_second_pass(struct drgn_dwarf_index *dindex,
 				file_name_hash = cu->file_name_hashes[decl_file - 1];
 			else
 				file_name_hash = 0;
-			if ((err = index_die(dindex, name,
-					     insn & DIE_FLAG_TAG_MASK,
+
+			struct drgn_dwarf_index_namespace *new_ns = NULL;
+			if ((err = index_die(ns, name,
+					     tag,
 					     file_name_hash, module,
-					     die_offset)))
+					     die_offset, &new_ns)))
 				return err;
+
+			if (new_ns) {
+				struct drgn_dwarf_index_pending_cu *pending;
+				#pragma omp critical(pending_cu_append)
+				{
```

Comment:
I believe the `{}` block is unnecessary. The `#pragma omp critical` can apply directly to the `if` statement.

Reply:
Oh, this is actually sort of a micro-optimization, I didn't want to acquire the lock (I'm assuming this is a lock) if new_ns is false. I could move this outside though if you think that won't matter, what do you think?

Reply:
I meant it could apply directly to the `if ((pending = ...))` statement. But now that I'm looking at it again, maybe it'd be better to pass in the `cu` to `index_die()`, and have `index_die()` add the `pending_cu` within the shard lock so we don't need an extra critical section and we don't need to return the namespace at all?

Reply:
sounds good, that's a lot nicer!